### PR TITLE
Create Meeting API- table contents

### PR DESCRIPTION
**Pull request by sushruth91**
Added repository, services and controller.

### CODE REVIEW

File: src/main/java/com/app/katchup/Meeting/MeetingController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class MeetingController {
+    @Autowired
+    MeetingService meetingService;
+
+    @PostMapping("/meeting")
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingController.java]: has to be meeting/create

File: src/main/java/com/app/katchup/Meeting/MeetingController.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    public ResponseEntity<Meeting> postMeeting(@RequestBody Meeting meeting){
+        meetingService.createMeeting(meeting);
+        return new ResponseEntity<>(meeting, HttpStatus.OK);
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingController.java]: HttpStatus.created

File: src/main/java/com/app/katchup/Meeting/MeetingRepository.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Repository
+public interface MeetingRepository extends CrudRepository<Meeting,String> {
+
+
+    @Query(value = "{'host' : ?0, 'startDateTime': ?2 , 'endDateTime':?3 , 'venue':?4}")
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingRepository.java]: parameter numbering wrong. please take care

File: src/main/java/com/app/katchup/Meeting/MeetingRepository.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    Meeting findIfHostHasDuplicateMeeting(String host, LocalDateTime startDateTime, LocalDateTime endDateTime, String venue);
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingRepository.java]: please rename to findMeetingByFilter

File: src/main/java/com/app/katchup/Meeting/MeetingService.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import com.app.katchup.MeetingResponse.model.MeetingResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MeetingService {
+    @Autowired
+    MeetingRepository meetingRepository;
+    public void createMeeting(Meeting meeting){
+        if(meetingRepository.findIfHostHasDuplicateMeeting(meeting.getHost(),meeting.getStartDateTime(),meeting.getEndDateTime(),
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingService.java]: please break it to two lines. update the method here based on the repository

Comment (sivajrm) [reply src/main/java/com/app/katchup/Meeting/MeetingService.java]: please watch for alignment in line#15

File: src/main/java/com/app/katchup/Meeting/MeetingService.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import com.app.katchup.MeetingResponse.model.MeetingInboxResponse;
+import com.app.katchup.MeetingResponse.model.MeetingResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MeetingService {
+    @Autowired
+    MeetingRepository meetingRepository;
+    public void createMeeting(Meeting meeting){
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingService.java]: update the return type from void to Meeting object

File: src/main/java/com/app/katchup/Meeting/MeetingController.java (hunk 2 above)
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingController.java]: get the returned meeting object from above line and return it along with responseentity

**Review by sivajrm — CHANGES_REQUESTED** [review]

File: src/main/java/com/app/katchup/Meeting/MeetingController.java (hunk 2 above)
Comment (sivajrm) [reply src/main/java/com/app/katchup/Meeting/MeetingController.java]: By default, you are returning meeeting object though the inside method returns some meeting for the host. This should not happen. You should get the response from the create meeting and check it if it is null, If it is null, return the error to the user if not return meeting created and the created object

File: src/main/java/com/app/katchup/Meeting/MeetingService.java (hunk 5 above)
Comment (sivajrm) [reply src/main/java/com/app/katchup/Meeting/MeetingService.java]: Return null if there exists a meeting already if not, go ahead and call save(), then get the same meeting to verify the insertion is successful and then return that object to the controller.

File: src/main/java/com/app/katchup/Meeting/MeetingService.java [hunk 7]
```diff
@@ -11,11 +11,14 @@
...
-    public void createMeeting(Meeting meeting){
-        if(meetingRepository.findIfHostHasDuplicateMeeting(meeting.getHost(),meeting.getStartDateTime(),meeting.getEndDateTime(),
+    public Meeting createMeeting(Meeting meeting){
+        if(meetingRepository.findMeetingByFilter(meeting.getHost(),
+                meeting.getStartDateTime(),meeting.getEndDateTime(),
                 meeting.getVenue())==null){
             meetingRepository.save(meeting);
         }
 
+        return meeting;
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingService.java]: if the condition is true, save it to the meeting and get the same meeting from DB then only the user will know the meeting-id and meeting pwd. 
if condition is false, return null

Comment (sivajrm) [reply src/main/java/com/app/katchup/Meeting/MeetingService.java]: Another option, put the application in debugger and show me the pwd and meeting id without getting the same meeting from DB. If this is possible, you dont need to do the DB fetching again.

File: src/main/java/com/app/katchup/Meeting/model/Meeting.java [hunk 8]
```diff
@@ -12,50 +12,47 @@
 import javax.validation.constraints.NotEmpty;
 import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Size;
+import javax.xml.ws.Response;
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/model/Meeting.java]: Where is this used ?

**Review by sivajrm — CHANGES_REQUESTED** [review]
Please take care of the comments of the latest commit

**Comment by sushruth91** [comment]
Updated return types

File: src/main/java/com/app/katchup/Meeting/model/Meeting.java [hunk 9]
```diff
@@ -3,57 +3,56 @@
...
+
+
+
+
+
+
+
 
-    @NotEmpty
-    private Status status;
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/model/Meeting.java]: What happenned to Status?

File: src/main/java/com/app/katchup/Meeting/model/Meeting.java (hunk 8 above)
Comment (sivajrm) [reply src/main/java/com/app/katchup/Meeting/model/Meeting.java]: please answer here

File: src/main/java/com/app/katchup/Meeting/MeetingController.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+@RestController
+public class MeetingController {
+    @Autowired
+    MeetingService meetingService;
+
+    @PostMapping("/meeting/create")
+    public ResponseEntity<Meeting> postMeeting(@RequestBody Meeting meeting){
+        Meeting meetingCreated=meetingService.createMeeting(meeting);
+        return new ResponseEntity<>(meetingCreated, HttpStatus.CREATED);
```
Comment (sivajrm) [inline src/main/java/com/app/katchup/Meeting/MeetingController.java:23]: please add null check for the returned object, if it is null, HttpStatus.Conflict(409)
else created

**Comment by sushruth91** [comment]
Added null checker.

**Comment by sushruth91** [comment]
Added null checker

**Review by sivajrm — APPROVED** [review]
With /meeting/create api

**Review by sivajrm — APPROVED** [review]

**Pull request merged**